overlay/generic/smartdc/lib/smartos_prompt_config.sh

```diff
@@ -271,55 +564,207 @@ promptpw()
+	for i in {1..80} ; do printf "-" ; done && printf "$newline"
+}
+
+print_warning()
+{
+	clear
+	printf "WARNING\n"
+	for i in {1..80} ; do printf "-" ; done && printf "\n"
+	printf "\n$1\n"
+
+	prmpt_str="\nPress [enter] to continue "
+	printf "$prmpt_str"
+	read continue
+}
+
+nicsup()
+{
+	[ $nicsup_done -eq 1 ] && return
+
+	local vlan_opts=""
+	if [[ "$admin_ip" == "dhcp" ]]; then
+		ifconfig $admin_iface up
+		ifconfig $admin_iface dhcp
+	else
+		ifconfig $admin_iface inet $admin_ip netmask $admin_netmask up
+	fi
+
+	if [[ -n ${external_nic} ]]; then
+		if [ -n "$external_vlan_id" ]; then
+			vlan_opts="-v $external_vlan_id"
+		fi
+
+		dladm create-vnic -l $external_iface $vlan_opts external0
+		ifconfig external0 plumb
+		ifconfig external0 inet $external_ip netmask $external_netmask up
+	fi
+
+	if [[ -n ${headnode_default_gateway}
+	    && ${headnode_default_gateway} != "none" ]]; then
+
+		route add default $headnode_default_gateway >/dev/null
+	fi
+
+	nicsup_done=1
+}
+
+nicsdown()
+{
+	ifconfig ${admin_iface} inet down unplumb
+	if [[ -n ${external_nic} ]]; then
+		ifconfig external0 inet down unplumb
+		dladm delete-vnic external0
+	fi
+}
+
+printdisklayout()
+{
+  json -f "$1" | json -e '
+    out = "vdevs:\n";
+    disklist = [];
+    for (var i = 0; i < vdevs.length; i++) {
+      var x = vdevs[i];
+      if (!x.type) {
+        out += "   " + x.name + "\n";
+        continue;
+      }
+      if (x.type === "mirror") {
+        out += "   " + x.type + "  " + x.devices[0].name + " " +
+          x.devices[1].name + "\n";
+        continue;
+      }
+      out += "   " + x.type + "\n";
+      var lout = "      ";
+      for (var j = 0; j < x.devices.length; j++) {
+        if ((lout + x.devices[j].name).length > 80) {
+          out += lout + "\n";
+          lout = "      " + x.devices[j].name + " ";
+        } else {
+          lout += x.devices[j].name + " ";
+        }
+      }
+      out += lout + "\n";
+    }
+    if (typeof (spares) !== "undefined" && spares.length > 0) {
+      out += "spares:\n";
+      var lout = "      ";
+      for (var i = 0; i < spares.length; i++) {
+        if ((lout + spares[i].name).length > 80) {
+          out += lout + "\n";
+          lout += "      " + spares[i].name + " ";
+        } else {
+          lout += spares[i].name + " ";
+        }
+      }
+      out += lout + "\n";
+    }
+    if (typeof (logs) !== "undefined" && logs.length > 0) {
+      out += "logs:\n";
+      var lout = "      ";
+      for (var i = 0; i < logs.length; i++) {
+        if ((lout + logs[i].name).length > 80) {
+          out += lout + "\n";
+          lout += "      " + logs[i].name + " ";
+        } else {
+          lout += logs[i].name + " ";
+        }
+      }
+      out += lout + "\n";
+    }
+    out += "total capacity:   " + Number(capacity / 1073741824).toFixed(2)
+      + " GB";
+    ' out
+}
 
+promptpool()
+{
+	local layout=""
+	local disks=
+	while [[ /usr/bin/true ]]; do
```

Comment:
true is typically built into bash and it can be more efficient to call it directly

``` bash
while true; do
end
```